src/NeoSharp.Core/Messaging/Handlers/AddrMessageMessageHandler.cs

```diff
@@ -10,55 +10,32 @@ public class AddrMessageMessageHandler : IMessageHandler<AddrMessage>
     {
         #region Variables
 
-        private readonly IPeerFactory _factory;
+        private readonly IServer _server;
 
         #endregion
 
         /// <summary>
         /// Constructor
         /// </summary>
-        /// <param name="factory">Factory</param>
-        public AddrMessageMessageHandler(IPeerFactory factory)
+        public AddrMessageMessageHandler(IServer server)
         {
-            _factory = factory ?? throw new ArgumentNullException(nameof(factory));
+            _server = server ?? throw new ArgumentNullException(nameof(server));
         }
 
-        public async Task Handle(AddrMessage message, IPeer sender)
+        public Task Handle(AddrMessage message, IPeer sender)
         {
-            // TODO: INJECT IServer
+            var connectedEndPoints = _server.ConnectedPeers
+                .Select(p => p.EndPoint)
+                .ToArray();
 
-            IPEndPoint[] peers = null;
+            var endPointsToConnect = message.Payload.Address
+                .Select(nat => nat.EndPoint)
+                .Where(ep => connectedEndPoints.Contains(ep) == false) 
```

Comment:
Use Equals or must be the same object?

Reply:
I have overrided Equals on EndPoint class.